settings.go

```diff
@@ -0,0 +1,179 @@
+package solus
+
+import (
+	"context"
+)
+
+// SettingsService handles all available methods with settings.
+type SettingsService service
+
+// Settings represents application settings.
+type Settings struct {
+	Hostname              string                        `json:"hostname"`
+	Mail                  SettingsMail                  `json:"mail"`
+	SendStatistics        bool                          `json:"send_statistics"`
+	LimitGroup            int                           `json:"limit_group"`
+	Registration          SettingsRegistration          `json:"registration"`
+	Theme                 SettingsTheme                 `json:"theme"`
+	NetworkRules          SettingsNetworkRules          `json:"network_rules"`
+	Notifications         SettingsNotifications         `json:"notifications"`
+	ComputeResource       SettingsComputeResource       `json:"compute_resource"`
+	NonExistentVMSRemover SettingsNonExistentVMSRemover `json:"non_existent_vms_remover"`
+	BillingIntegration    SettingsBillingIntegration    `json:"billing_integration"`
+	DNS                   SettingsDNS                   `json:"dns"`
+	Update                SettingsUpdate                `json:"update"`
+	Features              SettingsFeatures              `json:"features"`
+}
+
+// SettingsResponse represents response from settings end-point.
+type SettingsResponse struct {
+	Data Settings `json:"data"`
+}
+
+// SettingsUpdateRequest represents available properties for updating settings.
+type SettingsUpdateRequest struct {
+	Hostname              string                        `json:"hostname,omitempty"`
+	Mail                  SettingsMail                  `json:"mail,omitempty"`
+	SendStatistics        bool                          `json:"send_statistics,omitempty"`
+	LimitGroup            int                           `json:"limit_group,omitempty"`
+	Registration          SettingsRegistration          `json:"registration,omitempty"`
+	Theme                 SettingsTheme                 `json:"theme,omitempty"`
+	NetworkRules          SettingsNetworkRules          `json:"network_rules,omitempty"`
+	Notifications         SettingsNotifications         `json:"notifications,omitempty"`
+	ComputeResource       SettingsComputeResource       `json:"compute_resource,omitempty"`
+	NonExistentVMSRemover SettingsNonExistentVMSRemover `json:"non_existent_vms_remover,omitempty"`
+	BillingIntegration    SettingsBillingIntegration    `json:"billing_integration,omitempty"`
+	DNS                   SettingsDNS                   `json:"dns,omitempty"`
+	Update                SettingsUpdate                `json:"update,omitempty"`
+	Features              SettingsFeatures              `json:"features,omitempty"`
+}
+
+type settingsResponse struct {
```

Comment:
Please move this type closer to the methods where it used.

Reply:
OK